molecad/main.py

```diff
@@ -0,0 +1,98 @@
+import requests
+import time
+from typing import (
+    Any,
+    Iterable,
+    Optional,
+    TypeVar
+)
+
+from molecad.builder import (
+    build_url,
+    input_specification,
+    operation_specification
+)
+from molecad.sample_sync_requests import (
+    generate_ids,
+    chunked,
+    delay_iterations,
+    request_data_json
+)
+
+T = TypeVar("T")
+
+
+def join_w_comma(*args: object) -> str:
+    """
+    Formats all iterable arguments to a string with comma.
+    :param args: any arguments.
+    :return: string of comma-separated values without whitespaces.
+    """
+    return ",".join(map(str, args))
+
+
+def prepare_request(
+        domain: str,
+        namespace: str,
+        identifiers: Iterable[T],
+        operation: str,
+        output: str,
+        property_tags: Optional[list] = None
+) -> str:
+    """
+    Prepares all arguments to be passed to URL builder.
+    Mainly, joins all list arguments with comma to string.
+    # TODO
+    :param domain:
+    :param namespace:
+    :param identifiers: yielded value from
+    :param operation:
+    :param output:
+    :param property_tags:
+    :return:
+    """
+    joined_identifiers = join_w_comma(*identifiers)
+    input_spec = input_specification(domain, namespace, joined_identifiers)
+    if property_tags is None:
+        operation_spec = operation_specification(operation)
+    else:
+        joined_tags = join_w_comma(*property_tags)
+        operation_spec = operation_specification(operation, joined_tags)
+    url = build_url(input_spec, operation_spec, output)
+    return url
+
+
+def execute_request(url: str, params: dict[str, str]) -> dict[int, Any]:
```

Comment:
```suggestion
def execute_request(url: str, params: Dict[str, str]) -> Dict[int, Any]:
```

Так не пойдёт для py3.8

Reply:
у меня ^3.9 в pyproject.toml указан